aten/src/ATen/native/cuda/BinaryMiscBackwardOpsKernels.cu

```diff
@@ -20,7 +20,7 @@ void sigmoid_backward_kernel_cuda(TensorIterator& iter) {
 }
 
 void tanh_backward_kernel_cuda(TensorIterator& iter) {
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "tanh_backward_cuda", [&]() {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(ScalarType::Half, iter.dtype(), "tanh_backward_cuda", [&]() {
     gpu_kernel(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
```

Comment:
`using thrust_t = typename ztype_cuda<scalar_t>::thrust_t;`

Reply:
Should be resolved here: https://github.com/pytorch/pytorch/pull/38486/commits/c36a4480497346ef3a831673a50be5d316645e33. Thanks!